enzyme/tools/enzyme-tblgen/caching.cpp

```diff
@@ -27,10 +27,51 @@ os << "  bool need_" << name << " = false;\n";
       }
       os 
 << ";\n";
-    }
-    os 
+  }
+  os 
 << "  bool cache_" << name << " = cacheMode && overwritten_" << name << " && need_" << name << ";\n";
 }
+  
+std::string get_input_mat(const llvm::DagInit *ruleDag) {
```

Comment:
nit: we are `using namespace llvm` in this file, so lets not use the llvm namespace here for consistency.

```suggestion
std::string get_input_mat(const DagInit *ruleDag) {
```